### PR TITLE
Fix GMP build on ARM

### PR DESCRIPTION
**Pull request by raviqqe**

### CODE REVIEW

File: Formula/gmp.rb [hunk 1]
```diff
@@ -20,7 +20,10 @@ def install
     ENV.cxx11 if build.cxx11?
     args = %W[--prefix=#{prefix} --enable-cxx]
     args << "--build=core2-apple-darwin#{`uname -r`.to_i}" if build.bottle? && OS.mac?
-    args << "ABI=32" if Hardware::CPU.is_32_bit? && !OS.mac?
+    if OS.linux?
```
Comment (sjackman) [inline Formula/gmp.rb]: Please change this line to…
```ruby
unless OS.mac?
```

File: Formula/gmp.rb [hunk 2]
```diff
@@ -20,7 +20,10 @@ def install
...
+      args << "ABI=32" if !Hardware::CPU.arm? && Hardware::CPU.is_32_bit?
+      args << "ABI=standard" if Hardware::CPU.arm?
```
Comment (sjackman) [inline Formula/gmp.rb]: Is `ABI=standard` required on ARM? Does it compile fine without it?
Please change these lines to…
```ruby
if Hardware::CPU.arm?
  args << "ABI=standard"
else
  args << "ABI=32" if Hardware::CPU.is_32_bit?
end
```
or possibly
```ruby
args << "ABI=32" if Hardware::CPU.intel? && Hardware::CPU.is_32_bit? && !OS.mac?
```

**Comment by raviqqe** [comment]
Thank you for your advice.

Here is my [~/.cache/Homebrew/Logs/gmp/01.configure](https://gist.github.com/7d3580f5673494450a43e7f5503d4c5c) after `brew install gmp`.

```
$ uname -a
Linux localhost 3.14.0 #1 SMP PREEMPT Mon Jun 5 16:51:11 PDT 2017 armv7l armv7l armv7l GNU/Linux
$ cat /etc/os-release                        NAME="Ubuntu"
VERSION="17.10 (Artful Aardvark)"                 
ID=ubuntu                
ID_LIKE=debian           
PRETTY_NAME="Ubuntu Artful Aardvark (development branch)"                                            
VERSION_ID="17.10"       
HOME_URL="https://www.ubuntu.com/"                
SUPPORT_URL="https://help.ubuntu.com/"            
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"                                                  
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"                  
VERSION_CODENAME=artful  
UBUNTU_CODENAME=artful
```

And, your script works great on my environment. I pushed the new commit.

**Comment by sjackman** [comment]
Affects ARM. Does not affect Intel.
```
configure: error: ABI=32 is not among the following valid choices: standard
```
https://gist.github.com/raviqqe/7d3580f5673494450a43e7f5503d4c5c#file-01-configure-L18

File: Formula/gmp.rb [hunk 3]
```diff
@@ -20,7 +20,7 @@ def install
     ENV.cxx11 if build.cxx11?
     args = %W[--prefix=#{prefix} --enable-cxx]
     args << "--build=core2-apple-darwin#{`uname -r`.to_i}" if build.bottle? && OS.mac?
-    args << "ABI=32" if Hardware::CPU.is_32_bit? && !OS.mac?
+    args << "ABI=32" if Hardware::CPU.intel? && Hardware::CPU.is_32_bit? && !OS.mac?
```
Comment (sjackman) [inline Formula/gmp.rb]: Let's instead do:
```ruby
if OS.mac?
    args << "--build=core2-apple-darwin#{`uname -r`.to_i}" if build.bottle?
else
    args << "ABI=32" if Hardware::CPU.intel? && Hardware::CPU.is_32_bit?
end
```
Sorry for the back-and-forth.

**Comment by raviqqe** [comment]
Ping?

**Comment by sjackman** [comment]
Looks good to me. I'll merge it right after PR https://github.com/Linuxbrew/homebrew-core/pull/3308 is merged.

**Comment by sjackman** [comment]
Merged. Thanks for your contribution to Linuxbrew, @raviqqe!

**Comment by raviqqe** [comment]
Thank you too!

**Pull request merged**